File: leetcode/217_contains_duplicate/217_contains_duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
 var containsDuplicate = function(nums) {
    let duplicate = new Set([]);
    for (let i = 0; i < nums.length; i++) {
        if (duplicate.has(nums[i])) return true
        duplicate.add(nums[i])
    }    
    return false
};

let nums = [1,1,1,3,3,4,3,2,4,2];


console.log(containsDuplicate(nums));

File: leetcode/217_contains_duplicate/217_contains_duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        duplicate = set([])
        for num in nums:
            if (num in duplicate):
                return True
            duplicate.add(num)
        
        return False


if __name__ == "__main__":
    nums = [1,1,1,3,3,4,3,2,4,2] 
    sol = Solution().containsDuplicate(nums)
    print(sol)
